cookbooks/aws-parallelcluster-environment/recipes/init/restore_home_shared_data.rb

```diff
@@ -18,19 +18,65 @@
   # Restore the shared storage home data if it doesn't already exist
   # This is necessary to preserve any data in these directories that was
   # generated during the build of ParallelCluster AMIs after converting to
-  # shared storage and backed up to a temporary location previously
+  # shared storage and backed up to a temporary location previously.
   # Before removing the backup, ensure the data in the new home is the same
   # as the original to avoid any data loss or inconsistency. This is done
   # by using rsync to copy the data and diff to check for differences.
+  # The diff command excludes files that were originally in /home to focus on
+  # the newly synchronized files. This approach is necessary because the /home
+  # directory may contain pre-existing files such as slurm-*.out generated by
+  # running SLURM jobs and the automatically created lost+found directory.
   # Remove the backup after the copy is done and the data integrity is verified.
   bash "Restore /home" do
     user 'root'
     group 'root'
     code <<-EOH
+      # Generate a list of existing files and dirs in /home before the sync
+      find /home -mindepth 1 > /tmp/home_existing_files.txt
```

Comment:
Why mindepth=1?

Reply:
The -mindepth 1 is used to avoid including the /home directory itself. This will only list the files and subdirectories under the /home directory.